game.js

```diff
@@ -0,0 +1,90 @@
+class Game {
```

Comment:
A lot of tests which are really good to ensure that all the methods work with different types of inputs